src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -783,6 +805,34 @@ class WalletService(
             .flatMap { walletRepository.save(it.copy(status = WalletStatusDto.DELETED.toString())) }
             .map { LoggedAction(Unit, WalletDeletedEvent(walletId.value.toString())) }
 
+    fun patchWalletStateToError(
+        walletId: WalletId,
+        reason: String?
+    ): Mono<it.pagopa.wallet.documents.wallets.Wallet> {
+        logger.info("Patching wallet state to error for [{}]", walletId.value.toString())
+        return walletRepository
+            .findById(walletId.value.toString())
+            .switchIfEmpty { Mono.error(WalletNotFoundException(walletId)) }
+            .map { it.toDomain().error(reason) }
+            .flatMap {
+                when (it.status) {
+                    WalletStatusDto.ERROR -> Mono.just(it)
```

Comment:
here filtering should be done against transient statuses and then update wallet status to error, right?